# HG changeset patch
# User Pantxo Diribarne <pantxo.diribarne@gmail.com>
# Date 1423417200 -3600
#      Sun Feb 08 18:40:00 2015 +0100
# Node ID c17e1cefdbd316d59815a13187fa3fafde7dafeb
# Parent  0f557da98f5b774e0941ee9d0ff77f499932c8f9
Fix reordering graphics property children for hidden objects (bug #44211)

* graphics.in.h (children_property::do_set): check if children also list contains hidden chidren.

diff --git a/libinterp/corefcn/graphics.in.h b/libinterp/corefcn/graphics.in.h
--- a/libinterp/corefcn/graphics.in.h
+++ b/libinterp/corefcn/graphics.in.h
@@ -1711,44 +1711,52 @@ protected:
   {
     const Matrix new_kids = val.matrix_value ();
 
     octave_idx_type nel = new_kids.numel ();
 
     const Matrix new_kids_column = new_kids.reshape (dim_vector (nel, 1));
 
     bool is_ok = true;
+    bool add_hidden = true;
 
     if (! error_state)
       {
         const Matrix visible_kids = do_get_children (false);
 
         if (visible_kids.numel () == new_kids.numel ())
           {
             Matrix t1 = visible_kids.sort ();
             Matrix t2 = new_kids_column.sort ();
+            Matrix t3 = get_hidden ().sort ();
 
             if (t1 != t2)
               is_ok = false;
+            
+            if (t1 == t3)
+              add_hidden = false;
           }
         else
           is_ok = false;
 
         if (! is_ok)
           error ("set: new children must be a permutation of existing children");
       }
     else
       {
         is_ok = false;
         error ("set: expecting children to be array of graphics handles");
       }
 
     if (is_ok)
       {
-        Matrix tmp = new_kids_column.stack (get_hidden ());
+        Matrix tmp = new_kids_column;
+
+        if (add_hidden)
+          tmp.stack (get_hidden ());
 
         children_list.clear ();
 
         // Don't use do_init_children here, as that reverses the
         // order of the list, and we don't want to do that if setting
         // the child list directly.
 
         for (octave_idx_type i = 0; i < tmp.numel (); i++)
